internal/ioutil/tree.go

```diff
@@ -0,0 +1,58 @@
+package ioutil
+
+import (
+	"fmt"
+)
+
+const (
+	treeItemPrefix     = "├── "
+	treeItemPrefixLast = "└── "
+	subTreePrefix      = "│   "
+	subTreePrefixLast  = "    "
+)
+
+type Node struct {
+	Value    string
+	Children []*Node
+}
+
+func NewTree(value string) *Node {
+	node := Node{Value: value, Children: []*Node{}}
+	return &node
+}
+
+func (n *Node) Add(value string) *Node {
```

Comment:
Since we home brew the package `tree`, we can have functions specific to `notation` like
```go
func (n *Node) AddPair(key, value string) *Node {
    return n.Add(key + ": " + value)
}
```
so that code like
```go
sigNode.Add(fmt.Sprintf("media type: %s", signature.MediaType))
```
can be simplified as
```go
sigNode.AddPair("media type", signature.MediaType)
```